File: src/main/java/common/registries/RegistryFactory.java
package common.registries;

import java.util.HashMap;

public class RegistryFactory<T> {
    public HashMap<String, T> MAP = new HashMap<>();

    public void init(){}

    public void register(String name, T object){
        MAP.put( name , object);
    }

    public T getObject(String name){
        return MAP.getOrDefault(name, null);
    }
}

File: src/main/kotlin/common/registries/RegistryFactory.kt
package common.registries

import java.util.*

abstract class RegistryFactory<T> {
    var MAP = HashMap<String, T>()

    fun init() {}

    fun register(name: String, `object`: T) {
        MAP[name] = `object`
    }

    fun getObject(name: String): T {
        return (MAP as java.util.Map<String, T>).getOrDefault(name, null)
    }
}
